sklearn/preprocessing/tests/test_data.py

```diff
@@ -1500,13 +1506,37 @@ def test_one_hot_encoder_sparse():
     # test exception on wrong init param
     assert_raises(TypeError, OneHotEncoder(n_values=np.int).fit, X)
 
+
+def test_one_hot_encoder_error_on_negative():
+    # Negative numerical values in inputs should raise an exception
```

Comment:
That behaviour derives from the assumption that inputs are 0-based ordinal representation of categories (like output from LabelEncoder). It no longer applies...?